test_download_feature: bag has extra Image feature values not present in catalog

## Summary

\`tests/feature/test_features.py::TestFeatures::test_download_feature\` fails because the downloaded bag has *more* Image→feature pairs than the live catalog reports.

\`\`\`
assert catalog_feature_values == bag_feature_values
E   Extra items in the right set:
E     ('4DG', '4KT')
E     ('4DE', '4KT')
E     ('4DJ', '4KT')
E     ('4DC', '4KT')
\`\`\`

## Suspects

- Bag export pulls feature rows from a different scope than the dataset's catalog \`feature_values\` query. The bag walker may follow FK paths that pull in extra Image rows (feature rows belonging to Images NOT in this dataset).
- Snapshot vs. live drift — bag built at one snapshot, catalog comparison run at a later snapshot where some feature rows have been removed.
- Sibling of #116 (bag-export reach for Execution rows) but with the *opposite* sign — bag has too many rows here, too few there.

## Out of scope for PR #112

Pre-existing failure (PR #112's bag-builder fix addresses different reach gaps; this one is the over-reach side of the same family of contracts).

--- a/src/deriva_ml/dataset/dataset_bag.py
+++ b/src/deriva_ml/dataset/dataset_bag.py
@@ -639,6 +639,28 @@ class DatasetBag:
         target_col = table if isinstance(table, str) else table.name
         records = list(self._feature_cache.fetch_feature_records(target_col, feature_name))
 
+        # Filter to target rows that actually live in the bag. The bag
+        # walker can over-reach association tables (e.g. include
+        # Execution_Subject_Health rows whose Subject FK points at a
+        # Subject outside the dataset slice — see #126). A value whose
+        # target row isn't in the bag is dangling: drop it so the bag's
+        # feature_values matches what the bag actually contains.
+        try:
+            target_rids = {
+                r["RID"] for r in self.model.get_table_contents(target_col)
+            }
+        except Exception:
+            # If the target table is missing from the bag entirely,
+            # fall through with the unfiltered set — the cache fetch
+            # would have raised anyway. Defensive only.
+            target_rids = None
+        if target_rids is not None:
+            records = [
+                r
+                for r in records
+                if getattr(r, target_col, None) in target_rids
+            ]
+
         # Apply execution_rids filter (Python-side; the bag cache doesn't
         # have a server-side query layer to push this into). Empty list
         # short-circuits to an empty result.

